### PR TITLE
Features/bc classification module

### PR DESCRIPTION
**Pull request by IstiqomahUswatunH**
# Description

Make small modifications to the data module, add a lightning module for breast cancer classification using ResNet-18, and training pipeline with a single train, validation, and test step for a fast development run

- New Features:
1. Small modification to the data module
2. Adding ResNet-18 architecture using torch module
3. Create a lightning module for breast cancer classification using ResNet-18
4. Create Hydra config for data module + model
5. Create testing functions for the training pipeline

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [x] Ran tests using `pytest` to ensure the correctness of the changes.
- [x] Tested with a dummy dataset created using `temp_path_factory`.
- [x] Tested the modified data module.
- [x] Tested the training pipeline

**Test Configuration**:

- Firmware version:
- Hardware: Asus ExpertBook Intel(R) Core(TM) i5-10210U CPU @ 1.60GHz   2.11 GHz
- Toolchain: Visual Studio Code (VSCode)
- SDK: Windows 10

# Checklist:

- [x] Did you make sure title is self-explanatory and the description concisely explains the PR?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/data/bcc_datamodule.py [hunk 1]
```diff
@@ -88,6 +88,7 @@ def setup(self, stage: Optional[str] = None):
             )
             if len(self.data_train) == 0:
                 raise ValueError('Train dataset is empty.')
+        # print("check setup data_train:", self.data_train)
```
Comment (haritsahm) [inline src/data/bcc_datamodule.py:91]: remove unused debugging function

File: src/data/bcc_datamodule.py [hunk 2]
```diff
@@ -104,6 +105,8 @@ def setup(self, stage: Optional[str] = None):
                 )
                 if len(self.data_test) == 0:
                     raise ValueError('Test dataset is empty.')
+        # print("check setup data_val:", self.data_val)
```
Comment (haritsahm) [inline src/data/bcc_datamodule.py:108]: remove unused debugging function

File: src/data/bcc_datamodule.py [hunk 3]
```diff
@@ -113,6 +116,7 @@ def setup(self, stage: Optional[str] = None):
                 )
                 if len(self.data_predict) == 0:
                     raise ValueError('Predict dataset is empty.')
+            # print("check setup data_predict:", self.data_predict)
```
Comment (haritsahm) [inline src/data/bcc_datamodule.py:119]: remove unused debugging function

File: src/data/bcc_datamodule.py [hunk 4]
```diff
@@ -127,6 +131,7 @@ def train_dataloader(self):
             pin_memory=self.hparams.pin_memory,
             shuffle=True,
         )
+    print('check train_dataloader :', train_dataloader)
```
Comment (haritsahm) [inline src/data/bcc_datamodule.py:134]: remove unused debugging function

File: tests/conftest.py [hunk 5]
```diff
@@ -212,3 +212,54 @@ def cfg_physenet(cfg_physenet_global, tmp_path) -> DictConfig:
...
+            cfg.logger = None
+
+    return cfg
+
+
+@pytest.fixture(scope='function')
+def cfg_bcresnet18(cfg_bcresnet_global, tmp_path) -> DictConfig:
+    """Fixture for providing a specific configuration for Breast Cancer Classification with ResNet.
+
+    Args:
```
Comment (haritsahm) [inline tests/conftest.py:250]: remove the args and returns. The function is quite clear by name and functionality

File: tests/conftest.py [hunk 6]
```diff
@@ -212,3 +212,54 @@ def cfg_physenet(cfg_physenet_global, tmp_path) -> DictConfig:
     yield cfg
 
     GlobalHydra.instance().clear()
+
+
+@pytest.fixture(scope='package')
+def cfg_bcresnet_global() -> DictConfig:
+    """Fixture for providing a global configuration for Breast Cancer Classification with ResNet.
+
+    Returns:
```
Comment (haritsahm) [inline tests/conftest.py:221]: remov returns. The function is quite clear by name and functionality

**Review by haritsahm — CHANGES_REQUESTED** [review]
Please check and update the docstrings and comments

**Comment by IstiqomahUswatunH** [comment]
kindly review, please

**Review by haritsahm — APPROVED** [review]
LGTM

**Pull request merged**